### PR TITLE
WIP pQueue functions and tests

### PR DESCRIPTION
**Pull request by gwong5**

### CODE REVIEW

File: src/pQueue.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+export default class PQueue {
+  constructor() {
+    this.arr = []
+    this.index = 0
+  }
+
+  enqueue(element, priority) {
+
+    this.arr[this.index++] = {element, priority}
+    // comparePriority(a,b) => {
```
Comment (shakalee14) [inline src/pQueue.js:12]: take out commented code.

File: src/pQueue.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+export default class PQueue {
+  constructor() {
+    this.arr = []
+    this.index = 0
```
Comment (shakalee14) [inline src/pQueue.js:6]: why are you using this.index again? if  you're not using it, then take it out.

**Review by shakalee14 — COMMENTED** [review]
what is this.index++ doing? is it incrementing up the index through out the array?

File: src/pQueue.js (hunk 1 above)
Comment (shakalee14) [reply src/pQueue.js:12]: 12-16

File: src/pQueue.js (hunk 2 above)
Comment (gwong5) [reply src/pQueue.js:6]: It serves no purpose, thank you

**Pull request closed without merging**